lib/src/reverse_engineering/youtube_http_client.dart

```diff
@@ -179,24 +195,31 @@ class YoutubeHttpClient extends http.BaseClient {
     final url = Uri.parse(
         'https://www.youtube.com/youtubei/v1/$action?key=AIzaSyAO_FJ2SlqU8Q4STEHLGCilw_Y9_11qcW8');
 
-    return retry<JsonMap>(() async {
+    return retry<JsonMap>(this, () async {
       final raw = await post(url, body: json.encode(body));
+      if (_closed) throw HttpClientClosedException();
+
       return json.decode(raw.body);
     });
   }
 
   @override
-  void close() => _httpClient.close();
+  void close() {
+    _closed = true;
+    _httpClient.close();
+  }
 
   @override
   Future<http.StreamedResponse> send(http.BaseRequest request) {
+    if (_closed) throw HttpClientClosedException();
```

Comment:
I think it would be enough the check for _closed just here is this function is always called when a new request is made.